lessons/advanced/metaprogramming.md

```diff
@@ -130,6 +130,53 @@ def test do
 end
 ```
 
+## Debugging
+
+Okay, right now we know how to use `quote/2`, `unquote/1` and write macros. But what if you have a huge chunk of quoted code and want to understand it? In this case, you can use `Macro.to_string/2`. Take a look at this example:
+
+```elixir
+iex> Macro.to_string(quote(do: foo.bar(1, 2, 3)))
+"foo.bar(1, 2, 3)"
+```
+
+And when you want to look at the code generated by macros you can combine them with `Macro.expand/2` and `Macro.expand_once/2`, these functions expand macros in the given quoted code. The first may expand it several times, while the former - only once. For example, let's modify `unless` example from the previous section:
+
+```elixir
+defmodule OurMacro do
+  defmacro unless(expr, do: block) do
+    quote do
+      if !unquote(expr), do: unquote(block)
+    end
+  end
+end
+
+require OurMacro
+quoted = quote do
+  OurMacro.unless true, do: "Hi"
+end
+```
+
+```
+iex> quoted |> Macro.expand_once(__ENV__) |> Macro.to_string |> IO.puts
+if(!true) do
+  "Hi"
+end
+```
+
+But if we run the same code with `Macro.expand/2`, it's intriguing:
+
+```elixir
+iex> quoted |> Macro.expand(__ENV__) |> Macro.to_string |> IO.puts
+case(!true) do
+  x when x in [false, nil] ->
+    nil
+  _ ->
+    "Hi"
+end
+```
+
+You may remember that we've already mentioned that `if` is a macro in Elixir. Here you can see that it gets transformed to a `case` statement.
```

Comment:
How about:

You may recall that we've mentioned  `if` is a macro in Elixir, here we see it expanded into the underlying `case` statement.